packages/runtime/src/widgets/Progress.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState, useEffect } from "react";
+import { Progress as AntProgress } from "antd";
+import { WidgetRegistry } from "../registry";
+import type { EnsembleWidgetProps } from "../util/types";
+
+export type ProgressProps = {
+  display?: "linear" | "circular";
+  countdown?: number;
+  styles?: {
+    size?: number;
+    thickness?: number;
+    color?: string;
+  };
+} & EnsembleWidgetProps;
+
+const Progress: React.FC<ProgressProps> = (props) => {
+  const { display, countdown, styles } = props;
+
+  // Determine whether to use linear or circular progress based on the "display" prop
+  const isCircular = display === "circular";
+  const strokeWidth = isCircular && styles?.thickness ? styles.thickness : 6; // Use styles.thickness if available for circular display, otherwise default to 6
+
+  // Calculate the percentage based on the countdown value
+  const [percent, setPercent] = useState(countdown ? 0 : -1);
+
+  useEffect(
+    function settingUpInitialAnimation() {
+      // If countdown is present and greater than 0, calculate the target percent
+      if (countdown && countdown > 0) {
+        const targetPercent = 100;
+        let currentPercent = 0;
+        const interval = setInterval(
+          () => {
+            if (currentPercent < targetPercent) {
+              currentPercent += 1;
+              setPercent(currentPercent);
+            } else {
+              clearInterval(interval);
+            }
+          },
+          (countdown * 1000) / targetPercent
+        );
+        return () => clearInterval(interval);
+      } else {
+        setPercent(-1);
+      }
+    },
+    [countdown]
+  );
+
+  return (
+    <div
+      style={{
+        width: !isCircular && styles?.size ? `${styles.size}px` : "auto", // Set width only for circular display
```

Comment:
Easier to read if you have separate blocks for returning circular and linear instead of checking multiple times, e.g.

```
if (isCircular) {
  return <>...</>
}

return <>....</>
```